File: 05_simple_factory/java/SimpleFactoryWithout.java
public class SimpleFactoryWithout {

    public static void main(String[] args) {
        System.out.println(serviceLogistics(10));
        System.out.println(serviceLogistics(20));
    }

    public static CargoVolume serviceLogistics(int cargoVolume) {
        switch (cargoVolume) {
            case 10:
                return new CargoVolume("Truck 10", 6, "100.000");
            case 20:
                return new CargoVolume("Truck 20", 16, "1.000.000");
            default:
                return null;
        }
    }

}

class CargoVolume {
    private String name;
    private int doors;
    private String price;

    public CargoVolume(String name, int doors, String price) {
        this.name = name;
        this.doors = doors;
        this.price = price;
    }

    @Override
    public String toString() {
        return "CargoVolume [name=" + name + ", doors=" + doors + ", price=" + price + "]";
    }
}

File: 05_simple_factory/javascript/simple.factory.without.js
const serviceLogistics = (cargoVolume) => {
  switch (cargoVolume) {
    case 10:
      return {
        name: "Truck 10",
        doors: 6,
        price: "100.000đ ",
      };
    case 20:
      return {
        name: "Truck 20",
        doors: 16,
        price: "1.000.000đ ",
      };
  }
};

console.log(serviceLogistics(10));
console.log(serviceLogistics(20));
